widget_driver/doc/widget_drivers.md

```diff
@@ -0,0 +1,142 @@
+# WidgetDrivers
+
+Let's take a deep dive into `WidgetDrivers` 🤿 😃
+
+---
+
+## Background
+
+As mentioned in the [README](../), the `WidgetDriver` drives your widgets. Meaning that it provides all the dynamic data to the widget and tells it when to update.
+
+Whenever you have a widget that has some data that changes, then you need some way to drive that logic.
+
+Most Flutter apps probably use some common state management system like `Provider` or `BLoC`. And then in the build method of the widget they have some `BlocBuilder` or `context.watch<MyThing>()`
+
+Or probably you even have multiple of these, so something like:
+
+```dart
+BlocBuilder<BlocA, BlocAState>(
+    builder: (context, aState) {
+        BlocBuilder<BlocB, BlocBState>(
+            builder: (context, bState) {
+                // Some widget code
+            }
+        )
+    }
+)
+```
+
+In this example these two blocs are driving the widget. If they emit new states then the widget updates. And then the widget combines these states somehow and draws the correct widget.
+
+Now this might seem like a perfectly fine approach but it has one big issue. You are putting business logic in the widget.
+
+This breaks the separation of concern. The widget is a ui component. Its responsibility is to specify the UI. It should not specify UI and make business decisions.
+
+Another issue with this approach is that it makes testing tougher. In your `widgetTests` you now need to test that the correct widgets are on the screen, and **also** that the two blocs behave correctly together.
+
+And every time you test any widget which in some sub child contains these widgets, you will need to provide mock values for these specific blocs. (welcome to mocking hell 😬)
+
+A better approach would be to have the widget only depend on one `thing`. And this `thing` only provides fixed data to the widget, meaning that the widget can just consume the data without adding more logic on top of it.  
+
+And even better would be if this thing somehow would not need to be mocked by parent widgets when tested.
+
+## `WidgetDriver` to the rescue 🚙
+
+By design, the `WidgetDriver` framework gives your widget a single `driver`. It does not give you two, and it does not give you an option to create a `driver` yourself. It just provides your widgets with a `driver` property. And as soon as the driver gets new data, your widget will update and grab the latest version of that data.
```

Comment:
That last sentence "And as soon as the driver gets new data, your widget will update and grab the latest version of that data." is not true. When data on the driver changes that will not lead directly to an updated widget. The driver controls when data changes should also lead to a redraw of the widget.

Reply:
Yeah, that is true.. I oversimplified things here.. You need to also call the `notifyWidget()`.. Good input.. I will update this part and clarify it